src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2900,6 +2900,19 @@ void GraphOptimizer::MergeReorderAndTranspose(Graph &graph) {
         auto& outOrder = outBlockedDesc->getOrder();
 
         if (checkAscendingFinalOrder(transposeOrder, layoutOrder, inOrder, outOrder)) {
+            const auto parentInPlace = parentNode->getParentEdgeAt(0)->inPlace(Edge::LOOK_UP);
+            const auto& childEdges = childNode->getChildEdgesAtPort(0);
+            const auto childInPlace = std::any_of(childEdges.begin(), childEdges.end(), [](const EdgePtr& edge) {
+                return edge->inPlace(Edge::LOOK_DOWN);
+            });
+            bool isOptimizedReorder = !(parentInPlace && childInPlace);
+            // In case of non optimized reorder OneDNN primitive is used,
```

Comment:
I would say it is mostly not about oneDNN reorder itself, but rather about reorder node semantics in general.

Reply:
Agree, the comment is rewritten, please take a look